modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/services/HealthCheckConsensus.scala

```diff
@@ -91,37 +93,86 @@ abstract class HealthCheckConsensus[
 
   def startOwnRound(key: HealthCheckKey) =
     clusterStorage.getPeers.flatMap { peers =>
-      createRoundId.flatMap {
+      createRoundId.map(HealthCheckRoundId(_, selfId)).flatMap {
         startRound(key, peers, _, OwnRound)
       }
     }
 
-  def participateInRound(key: HealthCheckKey, roundId: RoundId): F[Unit] =
+  def participateInRound(key: HealthCheckKey, roundId: HealthCheckRoundId): F[Unit] =
     clusterStorage.getPeers.flatMap { peers =>
       startRound(key, peers, roundId, PeerRound)
     }
 
   def startRound(
     key: HealthCheckKey,
     peers: Set[Peer],
-    roundId: RoundId,
+    roundId: HealthCheckRoundId,
     roundType: HealthCheckRoundType
-  ): F[Unit] = ???
+  ): F[Unit] =
+    Clock[F].monotonic
+      .map(_.toSeconds)
+      .flatMap { startedAt =>
+        allRounds.get.map {
+          case ConsensusRounds(historical, inProgress) =>
+            def inProgressRound = inProgress.get(key)
+            def historicalRound = historical.find(_.roundId == roundId)
+
+            historicalRound.orElse(inProgressRound).map(_ => none).getOrElse {
+              HealthCheckRound.make[F, A, B]().some
```

Comment:
This will result in `Option[Option[A]]`. Is the nested option necessary? It makes reasoning about the code difficult to follow.

Reply:
Yes, I think so.
tl;dr;
It checks if historical or inProgress round exists, and if no - creates a new round declaration.
It returns the Option (to indicate whether new round should be created or no) and the value is `F[Round]`
so it's not `Option[Option[A]]` but `Option[F[A]]`

Reply:
Yep, you're right, the final type of this expression is `F[Option[F[A]]]`.